.github/workflows/test.yaml

```diff
@@ -57,6 +58,12 @@ jobs:
         uses: actions/checkout@v3
       - name: Unit test
         run: make test
+      - name: Initialize CodeQL
```

Comment:
can we add a parallel job for this? seems to take a long time

Reply:
I can try with a parallel job on my machine. 

The samples point to init, make, then analyse.
If a do it in parallel then then perhaps  I can skip the unit tests (make tests)